### PR TITLE
Ruby group capstone - Catalog of my things

### PR DESCRIPTION
**Pull request by SaneMyburg**
# Hi 👋
#### We created a catalogue of things as a group of 3 we: ####
- Created an Item class in a separate .rb file.
- Add all methods visible in the [diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png).
- Implemented methods:
    - can_be_archived?() in the Item class
    - move_to_archive() in the Item class
 - Created a main.rb file that will serve as your console app entry point.
 - Implemented startup actions:

### Team member #1 [Oussama](https://github.com/codedit334)
- Created a Book class in a separate .rb file.
- Created a Label class with an association to the Item class (in a separate .rb file).
- Implemented methods:
    - add_item method in the Label class
    - can_be_archived?() in the Book class
- Add unit tests for all implemented methods.
- The following options should be available:
    - List all books
    - List all labels (e.g. 'Gift', 'New')
    - Add a book
-  Preserved all data by saving collections in .json files.
- Created a schema.sql file with tables that will be analogical to the structure of the classes that were created.

### Team member #2 [Sane](https://github.com/SaneMyburg)
- Created MusicAlbum class in a separate .rb file.
- Created Genre class with an association to the Item class (in a separate .rb file).
- Implement methods:
    - add_item method in the Genre class
    - can_be_archived?() in the MusicAlbum class
 - Added unit tests for all implemented methods
 -  Preserved all data by saving collections in .json files.
 - Create a schema.sql file with tables that will be analogical to the structure of the classes that were created:

### Team member #3 [Nicholas](https://github.com/atok624)
- Created a Game class in a separate .rb file.
- Created an Author class with an association to the Item class (in a separate .rb file).
- Implement methods:
    - add_item method in the Author class
    - can_be_archived?() in the Game class
- Added unit tests for all implemented methods.
 - Preserved all data by saving collections in .json files.
 - Create a schema.sql file with tables that will be analogical to the structure of the classes that were created.

### CODE REVIEW

File: item.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require 'date'
+
```
Comment (1GORDON) [inline item.rb:2]: Great job team 🤝

- [ ] As stated in the project requirements, kindly ensure all associations (1-to-many relationships) are implemented. Kindly make sure the `Genre`, `Author`, and `Label` Classes have many items implemented. 
- Use this as an example of Genre to guide you.
genre

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @SaneMyburg, @codedit334, and @atok624 ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️ 
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️

- [ ] As stated in the project requirements, kindly record a video presenting the features of the project, also ensure all the members appear in the video and talk about the project and the video should contain a highlight about a piece of the code that you find interesting or proud of making.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: item.rb (hunk 1 above)
Comment (KDlamini) [reply item.rb:2]: Well Resolved ✔️ 

image

**Review by KDlamini — APPROVED** [review]
### Hi  team @SaneMyburg, @codedit334, @atok624 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the project requirements for the Catalog of My Things Ruby capstone. 🎊 🥳  
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!! 🎆 

## HIGHLIGHTS 🌟 

✔️ Good UI implementation
✔️ Good PR title and description
✔️ Class relationships and associations are included
✔️ Good file structure and code organization

## Video Presentation 🎥 

✔️  There is a recorded presentation video presenting the features of the project with a maximum length of 5 minutes.
✔️  The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
✔️  The video contains a demo of the project, highlighting all the project features.
✔️  The video contains a highlight about a piece of the code the student finds interesting or is proud of making.

# Status: Approved ✔️ 

oscars-standing-ovation
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**